e2e/tests/query.feature

```diff
@@ -0,0 +1,341 @@
+
+  Scenario: To query an existing account, wrong height
+	Given the user is running the network "single_node_network"
+	Given the user has a pocket client
+	When the user runs the command "query account {{address}} 5600000" against validator 0
+	Then the user should be able to see standard output containing "wanted to load target 5600000 but only found up to"
+
+  Scenario: To query an non-existing Account
+	Given the user is running the network "single_node_network"
+	Given the user has a pocket client
+	When the user runs the command "query account c7b6a62385f8999c0cb63ad9cb464ee730c597b2 0"
+	Then the user should be able to see standard output containing "null"
+
+  Scenario: To query an existing account, incomplete command
+	Given the user has a pocket client
+	When the user runs the command "query account"
+	Then the user should be able to see standard error containing "Usage:"
+
+  Scenario: To query an existing account txs
+	Given the user is running the network "single_node_network"
+	Given the user has a pocket client
+	When the user runs the command "query account-txs c7b6a62385f8999c0cb63ad9cb464ee730c597b1 0 1 0 0" against validator 0
+	Then the user should be able to see standard output containing "page_count"
+	And the user should be able to see standard output containing "total_txs"
+
+  Scenario: To query an existing account txs, wrong address
+	Given the user is running the network "single_node_network"
+	Given the user has a pocket client
+	When the user runs the command "query account-txs c7b6a62385f8999c0cb63ad9cb464ee730c597bw 0 1 0 0" against validator 0
+	Then the user should be able to see standard output containing "encoding/hex: invalid byte:"
+
+  Scenario: To query an existing account txs, incomplete command
+	Given the user has a pocket client
+	When the user runs the command "query account-txs"
+	Then the user should be able to see standard error containing "Usage:"
+
+  Scenario: To query current acl
+	Given the user is running the network "single_node_network"
+	Given the user has a pocket client
+	When the user runs the command "query acl"
+	Then the user should be able to see standard output containing "/v1/query/acl"
+	And the user should be able to see standard output containing "gov/non_map_acl"
+
+  Scenario Outline: Unsuccessful Query of an App
+	Given the user is running the network "single_node_network"
+	Given the user has a pocket client
+	When the user runs the command "query app {{address}} 0" against validator <index>
+	Then the user should be able to see standard output containing <message>
+	Examples:
+	  | index | message                      |
+	  | -1    | "invalid byte"               |
+	  | 0     | "application does not exist" |
+
+  Scenario: To query a non-existing app from address, address has incorrect length but only valid bytes
+	Given the user has a pocket client
+	When the user runs the command "query app 4920ce1d787123456aeff366c79e8aa2 0"
```

Comment:
Can we possibly make the app ID invalid as well (like giving it a value of 0 or 1 or -1, something impossible to be generated)? We don't want the possibility of a conflict is an app with the same id is created which would make the test fail (given that "4920ce1d787123456aeff366c79e8aa2" refers to the app ID"). Probability of that is low but with more usage, it is more likely - just to be safe.

Reply:
In this scenario the address has an incorrect length, so it's impossible to conflict with a correct app id.